### ISSUE TITLE
Inline node

### ISSUE THREAD

**Issue by Yomguithereal** [chjj/blessed#172]
Hello @chjj. Sorry if the question is naïve or if I did not read the documentation thoroughly enough but is there something looking like `inline` nodes in `blessed`, that would work a bit like an HTML `span`, for instance. The aim here would be to append several `text` nodes that would follow the flow of the text without needing to position them finely.

**Comment by chjj** [comment]
You can create a [`Layout` element][1] which will abide by whatever auto-positioning function you pass to it. It's basically a `Box` just with some special behavior happening to the child elements inside. There are two built-in renderers: `grid` and `inline`. Although `inline` is technically more like `inline-block` in css terms.

The only problem is you need some understanding of how blessed positioning works internally (which might be more complex than it needs to be) in order to write a renderer. I recommend looking at the example if you want to write your own.

[1]: https://github.com/chjj/blessed#layout-from-element

**Comment by dbkaplun** [comment]
You can also use `shrink:true` to have the element autocompute its dimensions. @chjj does this still work?

**Comment by chjj** [comment]
@dbkaplun, yes, that works (the newer way is `width: 'shrink', height: 'shrink'`), but the `Layout` element is what can provide real auto-positioning.

**Comment by Yomguithereal** [comment]
I'll try both those solutions if I don't find a way to merge react text nodes before I need to rely on strange inline elements. Thank you.
